lib/encoding/encoding_pure_test.go

```diff
@@ -0,0 +1,294 @@
+// +build !cgo
+
+package encoding
+
+import (
+	"math/rand"
+	"reflect"
+	"testing"
+)
+
+func TestIsConst(t *testing.T) {
```

Comment:
Leave common tests that work the same in pure Go and cgo-enabled builds in `encoding_test.go`, while move cgo-specific tests to `encoding_cgo_test.go` and pure Go-specific tests must be put in `encoding_pure_test.go`.